source/ui/input-text-model.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+
+#include "ui/input-text-model.hpp"
+#include "core/conversions.hpp"
+
+InputTextModel::InputTextModel(QQmlContext* context, MemoryAccess m): QObject(), context(context), start(0), maxLength(20), memoryAccess(m)
+{
+    context->setContextProperty("inputtextMod", this);
+}
+
+void InputTextModel::newText(QString text){
+    std::string stdText=text.toStdString();
+    //std::cout<<stdText<<std::endl;
+
+
+    for(unsigned i=0; i<stdText.length(); i++){
```

Comment:
use `size_t` or something. You generally dont' find primitive types that often in implementation files